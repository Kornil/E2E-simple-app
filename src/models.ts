export interface INote {
  id: string;
  text: string;
  xPosition: number;
  yPosition: number;
}
