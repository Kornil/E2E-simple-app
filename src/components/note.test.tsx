import React from "react";
import { render } from "@testing-library/react";

import Note from "./note";
import { INote } from "models";

jest.mock("uuid/v4", () => () => "test-id");

const mockNote: INote = {
  id: "test-id",
  yPosition: 100,
  xPosition: 100,
  text: "Hello World",
};

describe("<Note />", () => {
  it("matches snapshots", () => {
    const { container } = render(
      <Note note={mockNote} onEditNote={jest.fn()} onDeleteNote={jest.fn()} />
    );
    expect(container).toMatchSnapshot();
  });
});
