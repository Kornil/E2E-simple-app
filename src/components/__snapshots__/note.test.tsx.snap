// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Note /> matches snapshots 1`] = `
.emotion-0 {
  position: absolute;
  top: calc(100px - 12px);
  left: calc(100px - 12px);
  height: 24px;
  width: 24px;
  background-color: rgb(255,156,102);
  border: none;
  border-radius: 4px;
}

.emotion-0:hover {
  z-index: 1;
}

<div>
  <button
    class="emotion-0 emotion-1"
  />
</div>
`;
